openwisp_radius/api/serializers.py

```diff
@@ -102,7 +103,20 @@ class Meta:
         fields = '__all__'
 
 
+class RadiusOrganizationField(serializers.SlugRelatedField):
+    def get_queryset(self):
+        queryset = Organization.objects.all()
+        request = self.context.get('request', None)
+        if not request.user.is_superuser:
+            queryset = queryset.filter(pk__in=request.user.organizations_dict)
+        return queryset
+
+    def to_representation(self, obj):
+        return getattr(obj, 'pk')
+
+
 class RadiusBatchSerializer(serializers.ModelSerializer):
+    organization = RadiusOrganizationField(slug_field='slug')
```

Comment:
I think using the uuid was fine since we have all API endpoints doing that, if we do this change only here the API will be inconsistent, have you tought about this? I don't know if changing this for all the endpoints is the right thing to do now (we'de need to double check select_related is used when listing objects otherwise we'd generate 1 query for each object in the list for each API endpoint, that's my main concern)

Reply:
I think It's consistent with the other endpoints. 
All the APIs are taking organization input slug of the organization from the URL and the output is `uuid`; I have maintained the same, input is `slug` and output contains the `uuid`. Previously for this endpoint as well, it was the same, input was `slug` from URL and output for `pk`; I have only moved the slug from URL to post data.

Do you think it'll be better to use `pk` in post data?

Reply:
Thanks for the explanation, that's fine